lib/input_sanitizer/v2/types.rb

```diff
@@ -44,6 +44,8 @@ def call(value, options = {})
         raise InputSanitizer::ValueNotAllowedError.new(value)
       elsif value.blank? && (options[:allow_blank] == false || options[:required] == true)
         raise InputSanitizer::BlankValueError
+      elsif options[:regexp] && Regexp.new(options[:regexp]).match(value) == nil
```

Comment:
I would rather expect that `options[:regexp]` is already an object of type `Regexp` and not create a new one (and thus making passing a raw string like `"[a-b]"` a valid input for regexp sanitizer.

`options[:regexp].match(value).nil?` should do the trick